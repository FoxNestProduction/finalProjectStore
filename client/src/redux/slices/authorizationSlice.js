import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isUserAuthorized: false,
  token: null,
};

const authorizationSlice = createSlice({
  name: 'authorization',
  initialState,
  reducers: {
    setAuthorization(state, action) {
      state.isUserAuthorized = action.payload;
    },
    setToken: (state, action) => {
      state.token = action.payload;
    },
  },

});

export const {
  setAuthorization,
  setToken,
} = authorizationSlice.actions;

export default authorizationSlice.reducer;
