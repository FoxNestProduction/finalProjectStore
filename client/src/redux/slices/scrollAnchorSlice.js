import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  productsScrollAnchor: {},
};

const scrollAnchorSlice = createSlice({
  name: 'scrollAnchor',
  initialState,
  reducers: {
    setProductsScrollAnchor(state, action) {
      state.productsScrollAnchor = action.payload;
    },
  },
});

export const {
  setProductsScrollAnchor,
} = scrollAnchorSlice.actions;

export default scrollAnchorSlice.reducer;
