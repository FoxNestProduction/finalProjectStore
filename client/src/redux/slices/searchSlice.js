import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  search: [],
  key: '',
};

/* eslint-disable no-param-reassign */

const searchSlice = createSlice({
  name: 'search',
  initialState,
  reducers: {
    setSearch(state, action) { // eslint-disable-line no-shadow
      state.search = action.payload;
      state.key = action.payload;
    },
  },
});

export const { setSearch } = searchSlice.actions;

/* eslint-enable no-param-reassign */

export default searchSlice.reducer;
