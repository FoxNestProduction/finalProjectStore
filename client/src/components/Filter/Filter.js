import { Button, CardMedia, Stack, ToggleButton, Typography } from '@mui/material';
import React, { useEffect } from 'react';
import Box from '@mui/material/Box';
import Slider from '@mui/material/Slider';
import { useSelector, useDispatch } from 'react-redux';
import {
  stylesWrap,
  stylesTitle,
  stylesWrapTitle,
  stylesSlider,
  stylesBtn,
  stylesBtnReset,
  stylesSortBtn,
  stylesCategoryIconsWrap,
  stylesCategoryItem,
  stylesToggleButton,
} from './styles';
import { setFilter } from '../../redux/slices/filterSlice';
import { setSearch, setInputSearchValue } from '../../redux/slices/searchSlice';
import useAlert from '../../customHooks/useAlert';
import CustomAlert from '../Alert/Alert';

const Filter = () => {
  const dispatch = useDispatch();
  /* eslint-disable no-undef */
  const products = useSelector((state) => state.products.products);
  const [pizza, setPizza] = React.useState(sessionStorage.getItem('pizza') === 'true' || false);
  const [burgers, setBurgers] = React.useState(sessionStorage.getItem('burgers') === 'true' || false);
  const [sushi, setSushi] = React.useState(sessionStorage.getItem('sushi') === 'true' || false);
  const [salads, setSalads] = React.useState(sessionStorage.getItem('salads') === 'true' || false);
  const [pasta, setPasta] = React.useState(sessionStorage.getItem('pasta') === 'true' || false);
  const [sandwiches, setSandwiches] = React.useState(sessionStorage.getItem('sandwiches') === 'true' || false);
  const [bbqMeat, setBbqMeat] = React.useState(sessionStorage.getItem('bbqMeat') === 'true' || false);
  const [drink, setDrink] = React.useState(sessionStorage.getItem('drink') === 'true' || false);
  const [isTrending, setIsTrending] = React.useState(sessionStorage.getItem('isTrending') === 'true' || false);
  const [mostPopular, setMostPopular] = React.useState(sessionStorage.getItem('mostPopular') === 'true' || false);
  const [isHealthy, setIsHealthy] = React.useState(sessionStorage.getItem('isHealthy') === 'true' || false);
  const [isSupreme, setIsSupreme] = React.useState(sessionStorage.getItem('isSupreme') === 'true' || false);
  const defaultSliderValue = 15;
  const [valueSlider, setValueSlider] = React.useState(Number(sessionStorage.getItem('valueSlider')) || defaultSliderValue);

  const { alert, handleShowAlert, handleCloseAlert } = useAlert();

  const saveFilterToSessionStorage = () => {
    sessionStorage.setItem('pizza', pizza.toString());
    sessionStorage.setItem('burgers', burgers.toString());
    sessionStorage.setItem('sushi', sushi.toString());
    sessionStorage.setItem('salads', salads.toString());
    sessionStorage.setItem('pasta', pasta.toString());
    sessionStorage.setItem('sandwiches', sandwiches.toString());
    sessionStorage.setItem('bbqMeat', bbqMeat.toString());
    sessionStorage.setItem('drink', drink.toString());
    sessionStorage.setItem('isTrending', isTrending.toString());
    sessionStorage.setItem('mostPopular', mostPopular.toString());
    sessionStorage.setItem('isHealthy', isHealthy.toString());
    sessionStorage.setItem('isSupreme', isSupreme.toString());
    sessionStorage.setItem('valueSlider', valueSlider.toString());
  };

  useEffect(() => {
    saveFilterToSessionStorage(); // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [pizza, burgers, sushi, salads, pasta, sandwiches, bbqMeat,
    drink, isTrending, mostPopular, isHealthy, isSupreme, valueSlider]);

  const anchor = useSelector((state) => state.scrollAnchor.scrollAnchor);

  const marks = [
    {
      value: 0,
      label: '0$',
    },
    {
      value: 10,
      label: '10$',
    },
    {
      value: 20,
      label: '20$',
    },
    {
      value: 30,
      label: '30$',
    },
  ];

  const foodCategories = {
    burgers: `${burgers}`,
    pizza: `${pizza}`,
    sushi: `${sushi}`,
    salads: `${salads}`,
    pasta: `${pasta}`,
    sandwiches: `${sandwiches}`,
    bbqMeat: `${bbqMeat}`,
    drink: `${drink}`,
  };

  const filteredItemsByCatagory = products.filter((prod) => {
    const category = prod.filterCategories;
    const price = prod.currentPrice;
    return (Object.values(foodCategories).includes('true')
      ? (JSON.parse(foodCategories[category]) && price < valueSlider)
      : (price < valueSlider));
  });

  const filters = [
    { condition: mostPopular, filterFunc: (el) => el.rating > 4 },
    { condition: isTrending, filterFunc: (el) => el.isTrending },
    { condition: isHealthy, filterFunc: (el) => el.isHealthy },
    { condition: isSupreme, filterFunc: (el) => el.isSupreme },
  ];

  const filteredAndSortedItems = filters.reduce((items, filter) => {
    if (filter.condition) {
      return items.filter(filter.filterFunc);
    }
    return items;
  }, filteredItemsByCatagory);

  const handleApplyFilter = () => {
    if (filteredAndSortedItems.length === 0) {
      // eslint-disable-next-line no-undef,no-alert
      // alert('Nothing found :(');
      // console.log(e);
      handleShowAlert(true);
      dispatch(setFilter([]));
    } else {
      dispatch(setFilter(filteredAndSortedItems));
      dispatch(setSearch([]));
      dispatch(setInputSearchValue(''));
    }

    if (anchor) {
      // eslint-disable-next-line react/prop-types
      anchor.scrollIntoView({
        block: 'start',
      });
    }
  };

  const handleResetFilter = () => {
    dispatch(setFilter([]));
    setPizza(false);
    setBurgers(false);
    setSushi(false);
    setSalads(false);
    setPasta(false);
    setSandwiches(false);
    setBbqMeat(false);
    setDrink(false);
    setIsTrending(false);
    setMostPopular(false);
    setIsHealthy(false);
    setIsSupreme(false);
    setValueSlider(defaultSliderValue);
  };

  return (
    <Stack component="section" sx={stylesWrap}>
      <Stack component="div">
        <Stack component="div" sx={stylesWrapTitle}>
          <Typography component="h3" sx={stylesTitle}>
            Category
          </Typography>
          <Button
            sx={stylesBtnReset}
            onClick={handleResetFilter}
          >
            Reset
          </Button>
        </Stack>
        <Stack component="div" sx={stylesCategoryIconsWrap}>
          <Stack
            component="div"
            direction="row"
            gap={{ mobile: '10px', tablet: '9px', desktop: '13px' }}
            justifyContent={{ mobile: 'space-between', tablet: 'space-around', lgTablet: 'space-between' }}
            sx={{ width: '100%' }}
          >
            <ToggleButton
              sx={stylesToggleButton}
              value="pizza"
              selected={pizza}
              onChange={() => {
                setPizza(!pizza);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/pizza.png" alt="pizza" />
                <Typography component="p">Pizza</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="burgers"
              selected={burgers}
              onChange={() => {
                setBurgers(!burgers);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/burger.png" alt="burger" />
                <Typography component="p">Burgers</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="sushi"
              selected={sushi}
              onChange={() => {
                setSushi(!sushi);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/sushi.png" alt="sushi" />
                <Typography component="p">Sushi</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="salads"
              selected={salads}
              onChange={() => {
                setSalads(!salads);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/salad_2.png" alt="salads" />
                <Typography component="p">Salads</Typography>
              </Stack>
            </ToggleButton>
          </Stack>
          <Stack
            component="div"
            direction="row"
            gap={{ mobile: '10px', tablet: '9px', desktop: '13px' }}
            justifyContent={{ mobile: 'space-between', tablet: 'space-around', lgTablet: 'space-between' }}
            sx={{ width: '100%' }}
          >
            <ToggleButton
              sx={stylesToggleButton}
              value="pasta"
              selected={pasta}
              onChange={() => {
                setPasta(!pasta);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/pasta.png" alt="pasta" />
                <Typography component="p">Pasta</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="sandwich"
              selected={sandwiches}
              onChange={() => {
                setSandwiches(!sandwiches);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/sandwich.png" alt="sandwich" />
                <Typography component="p">Sandwich</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="bbqMeat"
              selected={bbqMeat}
              onChange={() => {
                setBbqMeat(!bbqMeat);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/bbqMeat.png" alt="bbq meat" />
                <Typography component="p">BBQ</Typography>
              </Stack>
            </ToggleButton>
            <ToggleButton
              sx={stylesToggleButton}
              value="drink"
              selected={drink}
              onChange={() => {
                setDrink(!drink);
              }}
            >
              <Stack component="div" sx={stylesCategoryItem}>
                <CardMedia component="img" image="./img/drink_2.png" alt="drink" />
                <Typography component="p">Drink</Typography>
              </Stack>
            </ToggleButton>
          </Stack>

          {/* <ToggleButton
            sx={stylesToggleButton}
            value="vegan"
            selected={vegan}
            onChange={() => {
              setVegan(!vegan);
            }}
          >
            <Stack sx={stylesCategoryItem}>
              <CardMedia component="img" image="./img/vegan.png" alt="vegan" />
              <Typography>Vegan</Typography>
            </Stack>
          </ToggleButton> */}
        </Stack>
      </Stack>
      <Stack component="div" sx={{ mt: { mobile: '25px', tablet: '30px' } }}>
        <Typography component="h3" sx={stylesTitle}>
          Sort By
        </Typography>
        <Stack component="div" direction="column" spacing={0} alignItems="flex-start" sx={{ width: '100%' }}>
          <Stack component="div" direction="row" justifyContent="space-between" sx={{ width: '100%' }}>
            <ToggleButton
              sx={stylesSortBtn}
              value="isTrending"
              selected={isTrending}
              onChange={() => {
                setIsTrending(!isTrending);
              }}
            >
              Trending
            </ToggleButton>
            <ToggleButton
              sx={stylesSortBtn}
              value="isHealthy"
              selected={isHealthy}
              onChange={() => {
                setIsHealthy(!isHealthy);
              }}
            >
              Healthy
            </ToggleButton>
          </Stack>

          <Stack component="div" direction="row" justifyContent="space-between" sx={{ width: '100%' }}>
            <ToggleButton
              sx={stylesSortBtn}
              value="isSupreme"
              selected={isSupreme}
              onChange={() => {
                setIsSupreme(!isSupreme);
              }}
            >
              Supreme
            </ToggleButton>
            <ToggleButton
              sx={stylesSortBtn}
              value="mostPopular"
              selected={mostPopular}
              onChange={() => {
                setMostPopular(!mostPopular);
              }}
            >
              Most Popular
            </ToggleButton>
          </Stack>
        </Stack>
      </Stack>

      <Stack component="div" sx={{ mt: { mobile: '15px', tablet: '10px', desktop: '15px' } }}>
        <Typography component="h3" sx={stylesTitle}>
          Price
        </Typography>
        <Box component="div" sx={{ width: '100%', mt: { mobile: '15px', tablet: '10px', desktop: '15px' } }}>
          <Slider
            sx={stylesSlider}
            max={30}
            aria-label="Always visible"
            // defaultValue={15}
            // getAriaValueText={valuetext}
            value={valueSlider}
            step={1}
            marks={marks}
            valueLabelDisplay="on"
            onChange={(event, newValue) => setValueSlider(newValue)}
          />
        </Box>
      </Stack>
      <Button
        sx={stylesBtn}
        onClick={(e) => handleShowAlert(e)}
      >
        Apply
      </Button>
      { alert && <CustomAlert type="warning" handleCloseAlert={handleCloseAlert} content="Nothing found :(" /> }
    </Stack>
  );
};

export default Filter;
