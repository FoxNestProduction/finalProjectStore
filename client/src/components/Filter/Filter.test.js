import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Filter from './Filter';
import { setFilterParams } from '../../redux/slices/filterSlice';
import { setIsApplyClicked } from '../../redux/slices/scrollAnchorSlice';
import { resetSearch } from '../../redux/slices/searchSlice';

const mockStore = configureStore();
const store = mockStore({
  filteredProducts: [],
  productsQuantity: null,
  loading: false,
  error: null,
  nothingFound: false,
  filterParams: {
    filterCategories: [],
    isTrending: false,
    rating: null, // mostPopular
    isHealthy: false,
    isSupreme: false,
    minPrice: 0,
    maxPrice: 30,
    sort: '',
    startPage: 1,
    perPage: 10,
  },
});

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('../../redux/slices/filterSlice', () => ({
  ...jest.requireActual('../../redux/slices/filterSlice'),
  setFilteredProducts: jest.fn(),
  setProductsQuantity: jest.fn(),
  setFilterParams: jest.fn(),
  setIsApplyClicked: jest.fn(),
  deleteFilteredData: jest.fn(),
  resetFilterParams: jest.fn(),
}));

jest.mock('../../redux/slices/searchSlice', () => ({
  ...jest.requireActual('../../redux/slices/searchSlice'),
  resetSearch: jest.fn(),
}));

describe('Filter component', () => {
  const filters = {
    filterCategories: ['pizza', 'burgers', 'sushi', 'salads'],
    isTrending: false,
    rating: null,
    isHealthy: false,
    isSupreme: false,
    minPrice: 0,
    maxPrice: 30,
  };

  test('renders Filter component', () => {
    render(<Filter filters={filters} setFilters={() => {}} resetFiltersLocalState={() => {}} />);

    expect(screen.getByText('Category')).toBeInTheDocument();
    expect(screen.getByText('Filter By')).toBeInTheDocument();
    expect(screen.getByText('Price')).toBeInTheDocument();
    expect(screen.getByText('Apply')).toBeInTheDocument();
  });

  test('handles category toggle buttons correctly', () => {
    render(<Filter filters={filters} setFilters={() => {}} resetFiltersLocalState={() => {}} />);
    const optionPizza = screen.getByText('Pizza');
    fireEvent.click(optionPizza);

    const optionBurgers = screen.getByText('Burgers');
    fireEvent.click(optionBurgers);

    const optionSushi = screen.getByText('Sushi');
    fireEvent.click(optionSushi);

    const optionSalads = screen.getByText('Salads');
    fireEvent.click(optionSalads);

    expect(optionPizza.closest('button')).toHaveClass('Mui-selected');
    expect(optionBurgers.closest('button')).toHaveClass('Mui-selected');
    expect(optionSushi.closest('button')).toHaveClass('Mui-selected');
    expect(optionSalads.closest('button')).toHaveClass('Mui-selected');
  });

  // test('handles "Apply Filter" button click correctly', async () => {
  //   render(
  //     <Provider store={store}>
  //       <Filter filters={filters} setFilters={() => {}} resetFiltersLocalState={() => {}} />
  //     </Provider>
  //   );
  //   fireEvent.click(screen.getByText('Apply'));
  //   await waitFor(() => {
  //     expect(setFilterParams).toHaveBeenCalledWith({
  //       filterCategories: ['pizza', 'burgers'],
  //       isHealthy: false,
  //       isSupreme: false,
  //       isTrending: false,
  //       maxPrice: 30,
  //       minPrice: 0,
  //       rating: null,
  //       startPage: 1,
  //     });
  //     expect(setIsApplyClicked).toHaveBeenCalledWith(true);
  //     expect(resetSearch).toHaveBeenCalled();
  //   });
  // });
});
