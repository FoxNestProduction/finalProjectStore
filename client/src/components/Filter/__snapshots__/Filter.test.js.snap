// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders Filter component 1`] = `
<DocumentFragment>
  <section
    class="MuiStack-root css-dpm171-MuiStack-root"
  >
    <div
      class="MuiStack-root css-nen11g-MuiStack-root"
    >
      <div
        class="MuiStack-root css-ar0lna-MuiStack-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-body1 css-11nk9k9-MuiTypography-root"
        >
          Category
        </h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1yb6wzo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Reset
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiStack-root css-1pjwift-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1geu7ea-MuiStack-root"
        >
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="pizza"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="pizza"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/pizza.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Pizza
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="burgers"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="burger"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/burger.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Burgers
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="sushi"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="sushi"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/sushi.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Sushi
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="salads"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="salads"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/salad_2.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Salads
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiStack-root css-1geu7ea-MuiStack-root"
        >
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="pasta"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="pasta"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/pasta.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Pasta
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="sandwiches"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="sandwich"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/sandwich.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Sandwich
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="bbqMeat"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="bbq meat"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/bbqMeat.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                BBQ
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1n7te43-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="drink"
          >
            <div
              class="MuiStack-root css-1sv8m96-MuiStack-root"
            >
              <img
                alt="drink"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="./img/drink_2.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Drink
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiStack-root css-h29emu-MuiStack-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-body1 css-11nk9k9-MuiTypography-root"
      >
        Filter By
      </h3>
      <div
        class="MuiStack-root css-1t5hxec-MuiStack-root"
      >
        <div
          class="MuiStack-root css-ee32gy-MuiStack-root"
        >
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1l39r7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="isTrending"
          >
            Trending
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1l39r7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="isHealthy"
          >
            Healthy
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiStack-root css-ee32gy-MuiStack-root"
        >
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1l39r7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="isSupreme"
          >
            Supreme
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-1l39r7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="mostPopular"
          >
            Most Popular
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiStack-root css-13o102m-MuiStack-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-body1 css-11nk9k9-MuiTypography-root"
      >
        Price
      </h3>
      <div
        class="MuiBox-root css-jelsty"
      >
        <span
          class="MuiSlider-root MuiSlider-marked MuiSlider-colorPrimary MuiSlider-sizeMedium css-618fq9-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 0%; width: 100%;"
          />
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="0"
            style="left: 0%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
            data-index="0"
            style="left: 0%;"
          >
            0$
          </span>
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="1"
            style="left: 33.333333333333336%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
            data-index="1"
            style="left: 33.333333333333336%;"
          >
            10$
          </span>
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="2"
            style="left: 66.66666666666667%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
            data-index="2"
            style="left: 66.66666666666667%;"
          >
            20$
          </span>
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="3"
            style="left: 100%;"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-markLabel MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
            data-index="3"
            style="left: 100%;"
          >
            30$
          </span>
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="0"
            style="left: 0%;"
          >
            <input
              aria-label="Always visible"
              aria-orientation="horizontal"
              aria-valuemax="30"
              aria-valuemin="0"
              aria-valuenow="0"
              data-index="0"
              max="30"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="0"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  0
                </span>
              </span>
            </span>
          </span>
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="1"
            style="left: 100%;"
          >
            <input
              aria-label="Always visible"
              aria-orientation="horizontal"
              aria-valuemax="30"
              aria-valuemin="0"
              aria-valuenow="30"
              data-index="1"
              max="30"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="30"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  30
                </span>
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-13mseku-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Apply
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
